Change let to const

DaengDo opened the pull request:
const로 객체 타입 값을 참조하여 기존의 call by value 를
call by reference 로 바꿔주었습니다.
기왕 객체로 만드는 김에 카운트를 증가시키는 함수도 메서드로 넣어줬습니다.
수업에서 보여준대로 중복되는 dom api를 함수로 묶어주었습니다.
appendChild() 를 append() 로 변경해주었습니다.

Commit:
Change let to const

const로 객체 타입 값을 참조하여 기존의 call by value 를
call by reference 로 바꿔주었습니다.
기왕 객체로 만드는 김에 카운트를 증가시키는 함수도 메서드로 넣어줬습니다.
수업에서 보여준대로 중복되는 dom api를 함수로 묶어주었습니다.
appendChild() 를 append() 로 변경해주었습니다.

hannut91 commented on src/index.js:
```diff
@@ -1,2 +1,37 @@
+//   render();
+// }
+
+const counter = {
+  count: 0,
+  handleClick() {
+    this.count += 1;
+    render();
+  },
+};
```
객체를 선언했을 때는, 객체 자체를 재할당 할 수는 없지만 객체에 속성들은 재할당이 가능하다는 것을 응용해서 문제를 해결해 주셨네요! 

아쉽게도 여전히 어떤 값이 변경된다는 사실에는 변함이 없습니다. 과제의 의도는 변수의 사용 대신 상수를 사용해서 문제를 해결하는 것인데요. 변수가 변한다는 사실 자체만으로도 프로그램의 복잡성을 많이 증가시킵니다. 어떤 값을 내가 사용할 때 이 값이 현재 어떤 값인지 예측하기 어렵기 때문이죠

hannut91 commented:
파이팅입니다 ㅎㅎ 도움이 필요하시면 말씀해 주세요

DaengDo committed:
Remove reassignment

어떤 값을 재할당하는 한다는 점에서 결국 객체로 참조하는건 똑같은
행위였군요

handleClick을 render 함수 안에서 사용하니 더 안전한 느낌도 듭니다.
이걸 함수 바깥의 변수를 참조하지 않았으니 이런걸 사이드 이펙트가
줄어들었다고 하나요?

또 검색해보고 querySelector보다 getElementById로 노드를 참조하는게
속도가 더 빠르다는걸 알게되었습니다.

저는 첫 과제부터 너무 어려웠습니다. 다른분들을 참고하고 검색을 통해 겨우
해결했습니다....ㅠ

hannut91 commented:
아 이게 닫혔군요
